src/network/networkd-dhcp6.c

```diff
@@ -35,6 +35,105 @@ static bool dhcp6_get_prefix_delegation(Link *link) {
                       RADV_PREFIX_DELEGATION_BOTH);
 }
 
+static bool dhcp6_has_preferred_subnet_id(Link *link) {
+        assert(link->network);
+        return link->network->router_prefix_subnet_id > -1;
+}
+
+static int dhcp6_get_preferred_delegated_prefix(
+                /* IN */
+                Manager* manager,
+                Link *link,
+                struct in6_addr *pd_prefix,
+                uint8_t pd_prefix_len,
+
+                /* OUT */
+                struct in6_addr *addr
```

Comment:
our typical way (at least for new code) for denoting what is input and output is to prefix the output with "ret_", and not prefix the input. hence addr → ret_addr. The comments then become unnecessary